aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePlugin.java

```diff
@@ -223,7 +225,22 @@ public void initialize(@NonNull Context context) throws AmplifyException {
                 initError, AmplifyException.TODO_RECOVERY_SUGGESTION
             );
         }
-        orchestrator.start();
+        startOrchestratorAsync();
+    }
+
+    private void startOrchestratorAsync() {
```

Comment:
To the fullest extent possible, we should keep this logic within the Orchestrator itself. The plugin is (theoretically) intended to be a thin veneer, and the Orchestrator itself has the job of managing start/stop lifecycle.

Is there a way to get the actions of the subscribe callbacks back into the Orchestrator, here?

Reply:
Yes...I was able to move this. The original thought was to have the orchestrator be controlled by the plugin, but it doesn't really buy us anything.